daemon/daemon.go

```diff
@@ -1142,6 +1142,14 @@ func (daemon *Daemon) GetRemappedUIDGID() (int, int) {
 // created. nil is returned if a child cannot be found. An error is
 // returned if the parent image cannot be found.
 func (daemon *Daemon) ImageGetCached(imgID string, config *runconfig.Config) (*image.Image, error) {
+	// for now just exit if imgID has no children.
+	// maybe parentRefs in graph could be used to store
+	// the Image obj children for faster lookup below but this can
+	// be quite memory hungry.
+	if !daemon.Graph().HasChildren(imgID) {
+		return nil, nil
+	}
+
```

Comment:
Sorry, missed your change to `HasChildren`, as it's later in the diff. This makes more sense now. I still think using `ByParent` would make more sense than `Map` in this function, but it's unrelated to your change.